lir/data/datasets/feature_data_csv.py

```diff
@@ -163,11 +180,61 @@ def _search_path(path: Path) -> Path:
         return path.resolve()
 
     def get_instances(self) -> FeatureData:
-        """Read the data from the csv file"""
-        if isinstance(self.path, Path):
-            path = self._search_path(self.path)
-            LOG.debug(f'parsing CSV file: {self.path} as {path}')
-            with open(path) as f:
-                return self._parse_file(f)
-        else:
-            return self._parse_file(self.path)
+        path = self._search_path(self.path)
+        LOG.debug(f'parsing CSV file: {self.path} as {path}')
+        with open(path) as f:
+            return self._parse_file(f)
+
+
+class FeatureDataCsvStreamParser(FeatureDataCsvParser):
+    def __init__(self, flo: IO, **kwargs: Any):
```

Comment:
```suggestion
    def __init__(self, fp: IO, **kwargs: Any):
```

in de python documentatie gebruiken ze voor file-like objects altijd `fp` (file pointer), bijvoorbeeld in [deze code](https://docs.python.org/3/library/argparse.html#fromfile-prefix-chars)